app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -9,5 +9,15 @@ def delete_resource(type, id, _data = {})
         action_result(true, "Deleting Cloud Volume #{cloud_volume.name}", :task_id => task_id)
       end
     end
+
+    def options
+      return super unless params[:ems_id]
+
+      klass = ManageIQ::Providers::StorageManager.find(params[:ems_id]).class.cloud_volumes_klass
```

Comment:
For OpenStack will the ems_id be the StorageManager here or will it be the CloudManager?

Reply:
It gets me `ManageIQ::Providers::Openstack::StorageManager::CinderManager`

Reply:
Hmm that's going to be an issue for Google then which like OpenStack has `CloudManager::CloudVolume` but it doesn't have a `StorageManager`

I can add this to the `BaseManager` and return `self::CloudVolume` and override it in amazon, but that depends on us getting the "right" ems_id back here.

Reply:
Yup, it's not returning anything for google:
```
  "resources": [
    {
      "href": "http://localhost:3000/api/providers/40",
      "id": "40",
      "name": "Amazon EBS Storage Manager",
      "type": "ManageIQ::Providers::Amazon::StorageManager::Ebs",
      "supports_block_storage": true
    },
    {
      "href": "http://localhost:3000/api/providers/5",
      "id": "5",
      "name": "OpenStack Cinder Manager",
      "type": "ManageIQ::Providers::Openstack::StorageManager::CinderManager",
      "supports_block_storage": true
    }
  ],

```

Reply:
This however means that there's no way to create a google cloud volume, only editing is allowed...

Reply:
Yeah exactly, I don't think checking supports_block_storage isn't exactly correct.  We should probably change the query to check for managers that support cloud_volumes that way we can assume that `ems.class::CloudVolume` will exist.

Reply:
:+1: